### PR TITLE
(2183) Remove no-longer-needed fields from Qualifications

### PR DESCRIPTION
**Pull request by Gweaton**
# Changes in this PR

Removes no-longer-needed fields from Qualifications:

- Qualification level
- Most common route to obtain
- Education duration
- Existence of professional experience required

We're now not displaying these Qualification fields anywhere e.g. on the
"show" pages or the Check your answers page when adding/editing a
profession.

I've had to update the seeds here, as we can no longer rely on a
qualification level to find the relevant qualification to set on a
profession, so I've gone for using the `routeToObtain` value here.

#### Qualification editing page

image

#### Check your answers

image

#### Public facing page

image

#### Qualification editing page

image

#### Check your answers

image

#### Public facing page

image

### CODE REVIEW

File: src/professions/professions.seeder.ts [hunk 1]
```diff
@@ -133,14 +133,7 @@ export class ProfessionsSeeder implements Seeder {
...
           qualification = await this.qualificationsRepository.save(
-            new Qualification(
-              qualification.level,
-              qualification.routesToObtain,
-              qualification.mostCommonRouteToObtain,
-              qualification.educationDuration,
-              qualification.mandatoryProfessionalExperience,
-              qualification.url,
-            ),
+            new Qualification(qualification.routesToObtain, qualification.url),
```
Comment (pezholio) [inline src/professions/professions.seeder.ts:136]: Do we even need the Qualifications relation now? Might simplify things to just have two fields in the Profession.

**Review by pezholio — APPROVED** [review]
Love to see some deleted code! This is good to go, but just one thing about qualifications. I'm tempted to say leave it until another time, but it might be good to do it now while we're touching this bit of the codebase. What do you think?

File: src/professions/professions.seeder.ts (hunk 1 above)
Comment (Gweaton) [reply src/professions/professions.seeder.ts:136]: There are some more fields like `ukRecognition` `otherCountriesRecognition` which are optional so we're not seeding them at the moment, so I think it still makes sense to have a separate Qualification (as much as I'd love to have fewer database entities).

Comment (pezholio) [reply src/professions/professions.seeder.ts:136]: Ah course!

**Pull request merged**